src/stdlib/distributions.jl

```diff
@@ -49,3 +49,49 @@ end
 function Distributions.logpdf(d::VecBinomialLogit{<:Real}, ks::Vector{<:Integer})
     return sum(logpdf_binomial_logit.(d.n, d.logitp, ks))
 end
+
+# OrderedLogistic distribution
+struct OrderedLogistic{T1, T2} <: DiscreteUnivariateDistribution
+   η::T1
+   cutpoints::Vector{T2}
+end
+
+# logpdf of the OrderedLogistic distribution
+function Distributions.logpdf(d::OrderedLogistic, k::Int)
+
+    K = length(d.cutpoints)+1
+
+    c =  d.cutpoints
+
+    if k==1
+        logp= log(logistic(c[k]-d.η))
+    elseif k<K
+        logp= log(logistic(c[k]-d.η) - logistic(c[k-1]-d.η))
+    else
+        logp= log(1-logistic(c[k-1]-d.η))
+    end
+
+    return(logp)
+end
+
+# rand of the OrderedLogistic distribution
+function Distributions.rand(d::OrderedLogistic)
+    cutpoints = d.cutpoints
+    η = d.η
+
+    if !issorted(cutpoints)
```

Comment:
Could this check be moved to construction-time? Would be nice to avoid having to check this each time that `rand` is called.

Reply:
This check is redundant now, when `Categorical` distribution will be used. In case the vector of cutpoints is not sorted, the error message will come from `Categorical`: `ArgumentError: Categorical: the condition isprobvec(p) is not satisfied.`

Reply:
I think we still want the check as the error message from `Categorical` is not informative.
Can we move the check before constructing `Categorical` maybe?

Reply:
Would it not be better to make a PR to distributions to fix the error message then? Doesn't seem like that code really belongs here if `Categorical` should be doing the checks for us.

Reply:
I don't think it's the case here. It just a special case here that when `issorted(cutpoints) == false`, `isprobvec(p)` is also false. For us we want to know where the error is from - it's because the cut points are not sorted. And `Categorical` should not be unaware of the reason why the passed-in `p` is not normalised.